Fix issue causing respository tests to fail

There is currently an issue with some repository tests failing depending on the order the tests are executed in. 

This appears to affect local machines but not Travis and specifically these tests

 * AdverseEventRepositoryTests
 * ComponentRepositoryTests
 * DonationBatchRepositoryTests
 * PostDonationCounsellingRepositoryTests

Exception seen in AdverseEventRepositoryTests  was:

```
Caused by: java.sql.SQLIntegrityConstraintViolationException: integrity constraint violation: foreign key no parent; FK_K8UIGMKEII4Q8WCAD9BJLV14 table: DONOR
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
	at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source)
	at org.hsqldb.jdbc.JDBCPreparedStatement.fetchResult(Unknown Source)
	at org.hsqldb.jdbc.JDBCPreparedStatement.executeUpdate(Unknown Source)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeUpdate(NewProxyPreparedStatement.java:147)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:208)
	... 76 more
Caused by: org.hsqldb.HsqlException: integrity constraint violation: foreign key no parent; FK_K8UIGMKEII4Q8WCAD9BJLV14 table: DONOR
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.Constraint.getException(Unknown Source)
	at org.hsqldb.Constraint.checkInsert(Unknown Source)
	at org.hsqldb.StatementDML.performIntegrityChecks(Unknown Source)
	at org.hsqldb.StatementDML.insertRowSet(Unknown Source)
	at org.hsqldb.StatementInsert.getResult(Unknown Source)
	at org.hsqldb.StatementDMQL.execute(Unknown Source)
	at org.hsqldb.Session.executeCompiledStatement(Unknown Source)
	at org.hsqldb.Session.execute(Unknown Source)
	... 80 more
```

--- a/test/repository/ComponentRepositoryTest.java
+++ b/test/repository/ComponentRepositoryTest.java
@@ -96,6 +96,7 @@ public class ComponentRepositoryTest {
 	public void after() throws Exception {
 		IDatabaseConnection connection = getConnection();
 		try {
+			clearSpringSecurityUser(); // reverse the initiation of the Spring Security user
 			IDataSet dataSet = getDataSet();
 			DatabaseOperation.DELETE_ALL.execute(connection, dataSet);
 		}
@@ -108,6 +109,10 @@ public class ComponentRepositoryTest {
 		BsisUserDetails user = new BsisUserDetails(userRepository.findUserById(1));
 		TestingAuthenticationToken auth = new TestingAuthenticationToken(user, "Credentials");
 		SecurityContextHolder.getContext().setAuthentication(auth);
+	}
+	
+	public void clearSpringSecurityUser() throws Exception {
+		SecurityContextHolder.getContext().setAuthentication(null);
 	}
 	
 	// TODO: test the following methods

--- a/test/repository/ComponentRepositoryTests.java
+++ b/test/repository/ComponentRepositoryTests.java
@@ -27,7 +27,6 @@ public class ComponentRepositoryTests extends ContextDependentTestSuite {
     private ComponentRepository componentRepository;
     
     @Test
-    @Ignore
     public void testUpdateComponentStatusForDonor_shouldOnlyUpdateMatchingComponents() {
         ComponentStatus oldStatus = ComponentStatus.AVAILABLE;
         ComponentStatus newStatus = ComponentStatus.UNSAFE;

--- a/test/repository/DonationBatchRepositoryTests.java
+++ b/test/repository/DonationBatchRepositoryTests.java
@@ -34,7 +34,6 @@ public class DonationBatchRepositoryTests {
     }
 
     @Test
-    @Ignore
     public void testCountOpenDonationBatches_shouldCountOpenAndNotDeletedBatches() {
         // Expected donation batches
         aDonationBatch().buildAndPersist(entityManager);

--- a/test/repository/PostDonationCounsellingRepositoryTests.java
+++ b/test/repository/PostDonationCounsellingRepositoryTests.java
@@ -46,8 +46,7 @@ public class PostDonationCounsellingRepositoryTests {
     private EntityManager entityManager;
     @Autowired
     private PostDonationCounsellingRepository postDonationCounsellingRepository;
-    
-    @Ignore
+
     @Test
     public void testFindDonorsFlaggedForCounsellingWithNoDates_shouldReturnDonorsFlaggedForCounselling() {
 
@@ -84,7 +83,6 @@ public class PostDonationCounsellingRepositoryTests {
         assertThat(returnedDonations, is(expectedDonations));
     }
     
-    @Ignore
     @Test
     public void testFindDonorsFlaggedForCounsellingWithDonorPanels_shouldReturnDonorsWithDonationsForDonorPanels() {
         
@@ -120,7 +118,6 @@ public class PostDonationCounsellingRepositoryTests {
         assertThat(returnedDonations, is(expectedDonations));
     }
 
-    @Ignore
     @Test
     public void testFindDonorsFlaggedForCounsellingWithStartDate_shouldReturnDonorsWithDonationsAferStartDate() {
         DateTime startDate = new DateTime().minusDays(7);
@@ -156,7 +153,6 @@ public class PostDonationCounsellingRepositoryTests {
         assertThat(returnedDonations, is(expectedDonations));
     }
 
-    @Ignore
     @Test
     public void testFindDonorsFlaggedForCounsellingWithEndDate_shouldReturnDonorsWithDonationsBeforeEndDate() {
         DateTime endDate = new DateTime().minusDays(7);
@@ -191,7 +187,6 @@ public class PostDonationCounsellingRepositoryTests {
         assertThat(returnedDonations, is(expectedDonations));
     }
 
-    @Ignore
     @Test
     public void testFindDonorsFlaggedForCounsellingWithDates_shouldReturnDonorsWithDonationsInDateRange() {
         DateTime startDate = new DateTime().minusDays(14);
@@ -227,8 +222,7 @@ public class PostDonationCounsellingRepositoryTests {
         
         assertThat(returnedDonations, is(expectedDonations));
     }
-    
-    @Ignore
+
     @Test(expected = NoResultException.class)
     public void testFindFlaggedPostDonationCounsellingForDonorWithNoPostDonationCounselling_shouldThrow() {
         
@@ -236,8 +230,7 @@ public class PostDonationCounsellingRepositoryTests {
 
         postDonationCounsellingRepository.findFlaggedPostDonationCounsellingForDonor(donor.getId());
     }
-    
-    @Ignore
+
     @Test
     public void testFindFlaggedPostDonationCounsellingForDonor_shouldReturnFirstFlaggedPostDonationCounsellingForDonor() {
         
@@ -283,8 +276,7 @@ public class PostDonationCounsellingRepositoryTests {
 
         assertThat(returnedPostDonationCounselling, is(expectedPostDonationCounselling));
     }
-    
-    @Ignore
+
     @Test
     public void testCountFlaggedPostDonationCounsellingsForDonorWithNoPostDonationCounsellings_shouldReturnZero() {
         
@@ -294,8 +286,7 @@ public class PostDonationCounsellingRepositoryTests {
         
         assertThat(returnedCount, is(0));
     }
-    
-    @Ignore
+
     @Test
     public void testCountFlaggedPostDonationCounsellingsForDonor_shouldReturnCorrectCount() {
         

